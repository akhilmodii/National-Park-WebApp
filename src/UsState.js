import React from 'react';


function UsState(props) {
  return (
    <div className="state-card">
      <h1>{props.name}</h1>
      <h3>Capital: <span>{props.selectedState.capitalCity}</span></h3>
      <h3>Population: <span>{props.selectedState.Population.toLocaleString()}</span></h3>
      <h3>Date Established: <span>{props.selectedState.dateEstablished}</span></h3>
      <h3>Number of Parks in State: <span>{props.selectedState.nationalParks.length}</span></h3>
      <h3>State Flower: <span>{props.selectedState.stateFlower}</span></h3>
    </div>
  )
}

export default UsState;