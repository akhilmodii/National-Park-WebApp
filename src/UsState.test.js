import React from 'react';
import UsState from './UsState';
import { shallow } from 'enzyme';

let name = "North Carolina";
let selectedState = {
  abbreviation: "NC",
  latitude: 35.630066,
  longitude: -79.806419,
  capitalCity: "Raleigh",
  dateEstablished: "Nov 21, 1789",
  Population: 10273419,
  stateFlower: "American Dogwood",
  totalAreaMi2: 53819,
  totalLandAreaMi2: 48618,
  totalWaterAreaMi2: 5201,
  nationalParks: ["Great Smoky Mountains"]
};

describe('UsState', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <UsState 
        name={name}
        selectedState={selectedState}
      />
    );
  });
  
  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});