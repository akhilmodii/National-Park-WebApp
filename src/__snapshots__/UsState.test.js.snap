// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsState should match the snapshot with all data passed in 1`] = `
<div
  className="state-card"
>
  <h1>
    North Carolina
  </h1>
  <h3>
    Capital: 
    <span>
      Raleigh
    </span>
  </h3>
  <h3>
    Population: 
    <span>
      10273419
    </span>
  </h3>
  <h3>
    Date Established: 
    <span>
      Nov 21, 1789
    </span>
  </h3>
  <h3>
    Number of Parks in State: 
    <span>
      1
    </span>
  </h3>
  <h3>
    State Flower: 
    <span>
      American Dogwood
    </span>
  </h3>
</div>
`;
